src/libs/auth.js

```diff
@@ -21,6 +21,28 @@ export const reloadAuthToken = () => {
   return getAuthInstance().currentUser.get().reloadAuthResponse().then(() => true, () => false)
 }
 
+export const hasBillingScope = () => {
+  return getAuthInstance().currentUser.get().hasGrantedScopes('https://www.googleapis.com/auth/cloud-billing')
+}
+
+/*
+ * Request Google Cloud Billing scope if necessary.
+ *
+ * NOTE: Requesting additional scopes may invoke a browser pop-up which the browser might block.
+ * If you use ensureBillingScope during page load and the pop-up is blocked, a rejected promise will
+ * be returned. In this case, you'll need to provide something for the user to deliberately click on
+ * and retry ensureBillingScope in reaction to the click.
+ */
+export const ensureBillingScope = async () => {
+  if (hasBillingScope()) {
+    return Promise.resolve()
+  } else {
+    const options = new window.gapi.auth2.SigninOptionsBuilder({ 'scope': 'https://www.googleapis.com/auth/cloud-billing' })
+    // Wait 100ms before doing the thing to avoid errors due to delays in applying the new scope grant
+    return getAuthInstance().currentUser.get().grant(options).then(() => new Promise(resolve => setTimeout(resolve, 100)))
```

Comment:
Since this function is `async`, we can leverage `await` instead of explicit promises. E.g.:

```
const ensureBillingScope = async () => {
  if (!hasBillingScope()) {
    await /* grant the scope */
    await Utils.delay(100)
  }
}
```

Reply:
TIL about `Utils.delay`.